include/detect_breakends/junction_detection.hpp

```diff
@@ -12,11 +12,13 @@
  *
  * \param alignment_file_path input file - path to the sam/bam file
  * \param insertion_file_path output file - path for the fasta file
+ * \param methods list of methods for detecting junctions (1: cigar_string, 2: split_read, 3: read_pairs, 4: read_depth)
  *
  * \details Detects junctions from the CIGAR strings and supplementary alignment tags of read alignment records.
  *          We sort out unmapped alignments, secondary alignments, duplicates and alignments with low mapping quality.
  *          Then, the CIGAR string of all remaining alignments is analyzed.
  *          For primary alignments, also the split read information is analyzed.
  */
 void detect_junctions_in_alignment_file(const std::filesystem::path & alignment_file_path,
-                                        const std::filesystem::path & insertion_file_path);
+                                        const std::filesystem::path & insertion_file_path,
+                                        const std::vector<uint8_t> methods);
```

Comment:
If I understand correctly, this function implements all four detection methods but can be parametrized by the `methods` parameter to only use a subset of methods. This makes a lot of sense because the alignment file has to be processed only once (in the best case) and different methods can be applied on the same read in one go.

However, this architecture is not really modularized, right? To be really modularized we would have four `detect_junctions`functions that each implement one method. But then, the alignment file would be processed four times in the worst case. So I guess that's why you decided to choose the other option?!

Reply:
Yes, that is correct. Maybe the term modularize is not really the right one.
The idea is, that we can combine different methods. And compare them. As we also want to compare run time, the complete decoupling would end in a higher run time if we combine two methods.
So I would leave it and would change the term modularisation with decoupling?

Reply:
I see. But I think it's not really a decoupling either. How about "Enable selection of methods for detecting junctions" as new title for this pull request?